mule-retry-api/src/main/java/org/mule/runtime/retry/api/policy/RetryPolicyTemplate.java

```diff
@@ -0,0 +1,155 @@
+   * Indicates if this policy is currently enabled or not.
+   *
+   * @return Whether {@code this} policy is enabled or not
+   * @since 4.3.0
+   */
+  default boolean isEnabled() {
+    return true;
+  }
+
+  /**
+   * Applies the retry policy in a non blocking manner by transforming the given {@code publisher} into one configured to apply
+   * the retry logic.
+   *
+   * @param publisher      a publisher with the items which might fail
+   * @param retryScheduler the scheduler to use when retrying. If empty, an internal reactor Scheduler will be used.
+   * @param <T>            the generic type of the publisher's content
+   * @return a {@link Publisher} configured with the retry policy.
+   * @since 4.2
+   */
+  default <T> Publisher<T> applyPolicy(Publisher<T> publisher, Scheduler retryScheduler) {
+    return createRetryInstance().applyPolicy(publisher, t -> true, t -> {
+    }, identity(), retryScheduler);
+  }
+
+  /**
+   * Applies the retry policy in a non blocking manner by transforming the given {@code publisher} into one configured to apply
+   * the retry logic.
+   *
+   * @param publisher     a publisher with the items which might fail
+   * @param shouldRetry   a predicate which evaluates each item to know if it should be retried or not
+   * @param onExhausted   an action to perform when the retry action has been exhausted
+   * @param errorFunction function used to map cause exception to exception emitted by retry policy.
+   * @param <T>           the generic type of the publisher's content
+   * @return a {@link Publisher} configured with the retry policy.
+   * @since 4.0
+   * @deprecated Use {@link #applyPolicy(Publisher, Predicate, Consumer, Function, Scheduler)} instead
+   */
+  @Deprecated
+  default <T> Publisher<T> applyPolicy(Publisher<T> publisher,
+                                       Predicate<Throwable> shouldRetry,
+                                       Consumer<Throwable> onExhausted, Function<Throwable, Throwable> errorFunction) {
+    return applyPolicy(publisher, shouldRetry, onExhausted, errorFunction, IMMEDIATE_SCHEDULER);
+
+  }
+
+  /**
+   * Applies the retry policy in a non blocking manner by transforming the given {@code publisher} into one configured to apply
+   * the retry logic.
+   *
+   * @param publisher      a publisher with the items which might fail
+   * @param shouldRetry    a predicate which evaluates each item to know if it should be retried or not
+   * @param onExhausted    an action to perform when the retry action has been exhausted
+   * @param errorFunction  function used to map cause exception to exception emitted by retry policy.
+   * @param retryScheduler the scheduler to use when retrying. If empty, an internal reactor Scheduler will be used.
+   * @param <T>            the generic type of the publisher's content
+   * @return a {@link Publisher} configured with the retry policy.
+   * @since 4.2
+   */
+  default <T> Publisher<T> applyPolicy(Publisher<T> publisher,
```

Comment:
this method uses reactivestreams, it should not be part of the api. Isn't this used only by the container?